apps/communications/dialer/js/telephony_helper.js

```diff
@@ -182,19 +168,15 @@ var TelephonyHelper = (function() {
       });
     }
     call.addEventListener('connected', onconnected);
-    call.ondisconnected = ondisconnected;
-    call.onerror = function errorCB(evt) {
-      if (onerror) {
-        onerror();
-      }
 
-      var errorName = evt.call.error.name;
+    call.addEventListener('disconnected', function handleDisconnect(evt) {
       loadTelephonyMessages(function() {
-        var messageType = emergencyOnly ? TelephonyMessages.NO_NETWORK :
-                                          TelephonyMessages.REGULAR_CALL;
-        TelephonyMessages.handleError(errorName, number, messageType);
+        TelephonyMessages.handleDisconnect(
+          evt.call.disconnectedReason, number, emergencyOnly
+        );
       });
-    };
+    });
+    call.addEventListener('disconnected', ondisconnected);
```

Comment:
It's awkward to add two event listeners like this. Combining them would be better.